api/management/commands/ingest_appeals.py

```diff
@@ -1,161 +1,192 @@
-                fields['event'] = event
-                fields['needs_confirmation'] = True
-
-            appeal, created = Appeal.objects.update_or_create(code=fields['code'], defaults=fields)
-            if created:
-                num_created = num_created + 1
+        atypes = {66: AppealType.DREF, 64: AppealType.APPEAL, 1537: AppealType.INTL}
+        atype = atypes[detail['APD_TYP_Id']]
+
+        if atype == AppealType.DREF:
+            # appeals are always fully-funded
+            amount_funded = detail['APD_amountCHF']
+        else:
+            amount_funded = 0 if detail['ContributionAmount'] is None else detail['ContributionAmount']
+
+        end_date = self.parse_date(detail['APD_endDate'])
+        # for new, open appeals, if we have a country, try to guess what emergency it belongs to.
+        # only consider emergencies within the past 90 days
+        event = None
+        if options['is_new_appeal'] and country is not None and end_date > datetime.utcnow().replace(tzinfo=timezone.utc):
+            six_mos = datetime.utcnow().replace(tzinfo=timezone.utc) - timedelta(days=90)
+            event = Event.objects.exclude(created_at__lt=six_mos).filter(countries__in=[country]).filter(dtype=dtype).order_by('-created_at')
+            if event.count():
+                event = event.first()
             else:
-                num_updated = num_updated + 1
+                event = None
+
+        fields = {
+            'aid': r['APP_Id'],
+            'name': r['APP_name'],
+            'dtype': dtype,
+            'atype': atype,
+
+            'country': country,
+            'region': region,
+
+            'sector': r['OSS_name'],
+            'code': r['APP_code'],
+            'status': {'Active': 0, 'Closed': 1, 'Frozen': 2, 'Archived': 3}[r['APP_status']],
+
+            'start_date': start_date,
+            'end_date': end_date,
+            'num_beneficiaries': detail['APD_noBeneficiaries'],
+            'amount_requested': detail['APD_amountCHF'],
+            'amount_funded': amount_funded,
+        }
+
+        if event is not None:
+            fields['event'] = event
+            fields['needs_confirmation'] = True
 
-        print('%s appeals created' % num_created)
-        print('%s appeals updated' % num_updated)
-        print('%s total appeals' % Appeal.objects.all().count())
+        return fields
+
+
+    def handle(self, *args, **options):
+        logger.info('Starting appeals ingest')
+        new, modified = self.get_new_or_modified_appeals()
+        logger.info('%s current appeals' % Appeal.objects.all().count())
+        logger.info('Creating %s new appeals' % len(new))
+        logger.info('Updating %s existing appeals that have been modified' % len(modified))
+
+        num_created = 0
+        for i, r in enumerate(new):
+            fields = self.parse_appeal_record(r, is_new_appeal=True)
+            try:
+                Appeal.objects.create(**fields)
+            except:
```

Comment:
I think you should log the actual error here too. It's possible that objects create fails for other reasons. Since you are only catching the error but not logging it, it will be difficult to debug.